### PR TITLE
fix(cdc): Fix CDC compliant check for class defined in Device descriptor

### PR DESCRIPTION
**Pull request by tore-espressif**
* Closes https://github.com/espressif/esp-usb/pull/89

### CODE REVIEW

File: host/class/cdc/usb_host_cdc_acm/host_test/main/descriptors/cypress_rfa.hpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      iInterface              0
+Device Status:     0x0000
+  (Bus Powered)
+  */
+
+namespace cypress_rfa {
+const uint8_t dev_desc[] = {
+    0x12, 0x01, 0x00, 0x02, 0x00, 0x00, 0x00, 0x08, 0xb4, 0x04, 0x03, 0x00, 0x00, 0x00, 0x01, 0x02, 0x00, 0x01
+};
+const uint8_t cfg_desc[] = {
```
[the remark this answers is not in the document]
Comment (tore-espressif) [reply host/class/cdc/usb_host_cdc_acm/host_test/main/descriptors/cypress_rfa.hpp:126]: I don't have these devices on my desk. These are CFG descriptors that users reported in various GH issues. So I cannot extend it...

File: host/class/cdc/usb_host_cdc_acm/host_test/main/test_parsing_cypress.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+using namespace cypress_rfa;
+
+SCENARIO("Custom descriptor parsing: Cypress", "[cypress][rfa]")
+{
+    GIVEN("RFA") {
+        const usb_device_desc_t *dev = (const usb_device_desc_t *)dev_desc;
+        const usb_config_desc_t *cfg = (const usb_config_desc_t *)cfg_desc;
+
+        SECTION("Interface 0") {
```
[the remark this answers is not in the document]
Comment (tore-espressif) [reply host/class/cdc/usb_host_cdc_acm/host_test/main/test_parsing_cypress.cpp:23]: The third interface is Vendor specific with 0 endpoints. I can add a test that will check that the last interface cannot be opened as CDC-ACM device

**Review by peter-marcisovsky — APPROVED** [review]
Changes LGTM.

File: host/class/cdc/usb_host_cdc_acm/host_test/main/test_parsing_cypress.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (tore-espressif) [reply host/class/cdc/usb_host_cdc_acm/host_test/main/test_parsing_cypress.cpp:23]: added :) will merge after CI passes

**Pull request merged**